tools/restatectl/src/connection.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{cmp::Ordering, collections::HashMap, fmt::Display, future::Future, sync::Arc};
+
+use cling::{prelude::Parser, Collect};
+use itertools::{Either, Itertools};
+use rand::{seq::SliceRandom, thread_rng};
+use tokio::sync::{Mutex, MutexGuard};
+use tonic::{codec::CompressionEncoding, transport::Channel, Response, Status};
+use tracing::debug;
+
+use restate_core::protobuf::node_ctl_svc::{
+    node_ctl_svc_client::NodeCtlSvcClient, GetMetadataRequest, IdentResponse,
+};
+use restate_types::{
+    logs::metadata::Logs,
+    net::AdvertisedAddress,
+    nodes_config::{NodesConfiguration, Role},
+    protobuf::common::{MetadataKind, NodeStatus},
+    storage::{StorageCodec, StorageDecode, StorageDecodeError},
+    Version, Versioned,
+};
+
+use crate::util::grpc_channel;
+
+#[derive(Clone, Parser, Collect, Debug)]
+pub struct ConnectionInfo {
+    // todo: rename this to be a node address for reusability across commands
+    /// Cluster Controller address
+    #[clap(
+        long="address",
+        value_hint = clap::ValueHint::Url,
+        default_value = "http://localhost:5122/",
+        env = "RESTATE_ADDRESS",
```

Comment:
How do I specify multiple addresses? `--address=[a, b, c]` or `--address=a --address=b`? Wondering whether this should be plural if it is the former.

Reply:
it's specified like `--address a --address b ...` This is why I preferred the singular form. I will update the doc string to denote that it can be supplied more than one time.

Reply:
When I run `restatectl --help` then it prints:

```
      --address <ADDRESSES>
          Specify the node address. This option can be used multiple times
```

where `ADDRESSES` is a plural.

This made me think that I can also provide multiple addresses to the `--address` option. However, this would only work if we add `value_delimiter = ','` to the clap attribute. Or maybe there is a way to make it a singular noun?

Reply:
Makes sense. I think i will restore the name and add the delimiter back. It will also make the env var work with multiple comma separated list of addresses as well.